# Checkpoint 1

## Carregar módulos

In [273]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

## Carregar dados

In [274]:
path = 'https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/time_series_covid19_confirmed_global.csv'
path_dead =  'https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/time_series_covid19_deaths_global.csv'

In [275]:
dados_paises_covid = pd.read_csv(path)
dados_paises_covid_dead = pd.read_csv(path_dead)

In [276]:
dados_paises_covid

Province/State        Country/Region        Lat        Long  1/22/20  \
0              NaN           Afghanistan  33.939110   67.709953        0   
1              NaN               Albania  41.153300   20.168300        0   
2              NaN               Algeria  28.033900    1.659600        0   
3              NaN               Andorra  42.506300    1.521800        0   
4              NaN                Angola -11.202700   17.873900        0   
..             ...                   ...        ...         ...      ...   
284            NaN    West Bank and Gaza  31.952200   35.233200        0   
285            NaN  Winter Olympics 2022  39.904200  116.407400        0   
286            NaN                 Yemen  15.552727   48.516388        0   
287            NaN                Zambia -13.133897   27.849332        0   
288            NaN              Zimbabwe -19.015438   29.154857        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  2/28/23  3/1/23  \
0          0        0        0        0        0  ...   209322  209340   
1          0        0        0        0        0  ...   334391  334408   
2          0        0        0        0        0  ...   271441  271448   
3          0        0        0        0        0  ...    47866   47875   
4          0        0        0        0        0  ...   105255  105277   
..       ...      ...      ...      ...      ...  ...      ...     ...   
284        0        0        0        0        0  ...   703228  703228   
285        0        0        0        0        0  ...      535     535   
286        0        0        0        0        0  ...    11945   11945   
287        0        0        0        0        0  ...   343012  343012   
288        0        0        0        0        0  ...   263921  264127   

     3/2/23  3/3/23  3/4/23  3/5/23  3/6/23  3/7/23  3/8/23  3/9/23  
0    209358  209362  209369  209390  209406  209436  209451  209451  
1    334408  334427  334427  334427  334427  334427  334443  334457  
2    271463  271469  271469  271477  271477  271490  271494  271496  
3     47875   47875   47875   47875   47875   47875   47890   47890  
4    105277  105277  105277  105277  105277  105277  105288  105288  
..      ...     ...     ...     ...     ...     ...     ...     ...  
284  703228  703228  703228  703228  703228  703228  703228  703228  
285     535     535     535     535     535     535     535     535  
286   11945   11945   11945   11945   11945   11945   11945   11945  
287  343079  343079  343079  343135  343135  343135  343135  343135  
288  264127  264127  264127  264127  264127  264127  264276  264276  

[289 rows x 1147 columns]

## Ajustes



- Remover países sem sentido/inexistentes

In [277]:
lista_paises = dados_paises_covid['Country/Region'].tolist()

# print(lista_paises)

# usarei o chatgpt para verificar o que não é um pais, pois a lista predefinida que utilizei
# em testes anteriores elimina paises como "Korean, north" e esquecia de outros como "US"

lista_nao_paises = ["Antarctica","Diamond Princess","Holy See","Kosovo","MS Zaandam","Summer Olympics 2020","Taiwan*","West Bank and Gaza","Winter Olympics 2022"]
dados_paises_covid = dados_paises_covid[~dados_paises_covid['Country/Region'].isin(lista_nao_paises)]

In [305]:
# Testes / Para lidar com erros, use o Run all ao inves de Run cell

# print(lista_paises)
# print(lista_nao_paises)
# print(dados_paises_covid.columns)
# print(dados_paises_covid['Country/Region'].unique())

- Nomes das colunas

In [279]:
dados_paises_covid.rename(columns={"Province/State": "province",
                                   "Country/Region": "country"},
                          inplace=True)

C:\Users\zrkvi\AppData\Local\Temp\ipykernel_18308\373085972.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dados_paises_covid.rename(columns={"Province/State": "province",


In [280]:
dados_paises_covid_dead.rename(columns={"Province/State": "province",
                                        "Country/Region": "country"},
                               inplace=True)

In [281]:
dados_paises_covid_dead.columns

Index(['province', 'country', 'Lat', 'Long', '1/22/20', '1/23/20', '1/24/20',
       '1/25/20', '1/26/20', '1/27/20',
       ...
       '2/28/23', '3/1/23', '3/2/23', '3/3/23', '3/4/23', '3/5/23', '3/6/23',
       '3/7/23', '3/8/23', '3/9/23'],
      dtype='object', length=1147)

- Reajustar dados

In [282]:
dados_paises_covid = dados_paises_covid.melt(id_vars=['province', 'country', 'Lat', 'Long'],
                                             var_name = 'date',
                                             value_name = 'infected')

In [283]:
dados_paises_covid_dead = dados_paises_covid_dead.melt(id_vars=['province', 'country', 'Lat', 'Long'],
                                                       var_name = 'date',
                                                       value_name = 'dead')

In [284]:
dados_paises_covid_dead.tail()

province               country        Lat        Long    date  dead
330322      NaN    West Bank and Gaza  31.952200   35.233200  3/9/23  5708
330323      NaN  Winter Olympics 2022  39.904200  116.407400  3/9/23     0
330324      NaN                 Yemen  15.552727   48.516388  3/9/23  2159
330325      NaN                Zambia -13.133897   27.849332  3/9/23  4057
330326      NaN              Zimbabwe -19.015438   29.154857  3/9/23  5671

In [285]:
dados_paises_covid_dead.dtypes

province     object
country      object
Lat         float64
Long        float64
date         object
dead          int64
dtype: object

- Ajustar datas

In [286]:
dados_paises_covid['date'] = pd.to_datetime(dados_paises_covid['date'])

C:\Users\zrkvi\AppData\Local\Temp\ipykernel_18308\1364628153.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  dados_paises_covid['date'] = pd.to_datetime(dados_paises_covid['date'])


In [287]:
dados_paises_covid_dead['date'] = pd.to_datetime(dados_paises_covid_dead['date'])

C:\Users\zrkvi\AppData\Local\Temp\ipykernel_18308\2926324975.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  dados_paises_covid_dead['date'] = pd.to_datetime(dados_paises_covid_dead['date'])


In [288]:
dados_paises_covid.tail()

province    country        Lat        Long       date  infected
320035      NaN  Venezuela   6.423800  -66.589700 2023-03-09    552162
320036      NaN    Vietnam  14.058324  108.277199 2023-03-09  11526994
320037      NaN      Yemen  15.552727   48.516388 2023-03-09     11945
320038      NaN     Zambia -13.133897   27.849332 2023-03-09    343135
320039      NaN   Zimbabwe -19.015438   29.154857 2023-03-09    264276

In [289]:
dados_paises_covid_dead.tail()

province               country        Lat        Long       date  dead
330322      NaN    West Bank and Gaza  31.952200   35.233200 2023-03-09  5708
330323      NaN  Winter Olympics 2022  39.904200  116.407400 2023-03-09     0
330324      NaN                 Yemen  15.552727   48.516388 2023-03-09  2159
330325      NaN                Zambia -13.133897   27.849332 2023-03-09  4057
330326      NaN              Zimbabwe -19.015438   29.154857 2023-03-09  5671

In [290]:
dados_paises_covid_dead.dtypes

province            object
country             object
Lat                float64
Long               float64
date        datetime64[ns]
dead                 int64
dtype: object

- Combinar os datasets

In [291]:
dados_paises_covid.shape

(320040, 6)

In [292]:
dados_paises_covid_dead.shape

(330327, 6)

In [293]:
dados_covid = pd.merge(dados_paises_covid, dados_paises_covid_dead,
                       on = ['province', 'country',	'Lat',	'Long',	'date'],
                       how = 'inner')

In [294]:
dados_covid.head()

province      country       Lat       Long       date  infected  dead
0      NaN  Afghanistan  33.93911  67.709953 2020-01-22         0     0
1      NaN      Albania  41.15330  20.168300 2020-01-22         0     0
2      NaN      Algeria  28.03390   1.659600 2020-01-22         0     0
3      NaN      Andorra  42.50630   1.521800 2020-01-22         0     0
4      NaN       Angola -11.20270  17.873900 2020-01-22         0     0

In [295]:
dados_covid.shape

(320040, 7)

## Aqui começa o trabalho

## Questão 1

In [299]:
dados_covid_filtrado = dados_covid[dados_covid['country'].isin(paises_validos)]
print(dados_covid_filtrado)
dados_covid_filtrado['country'].nunique()

       province      country        Lat        Long       date  infected  \
0           NaN  Afghanistan  33.939110   67.709953 2020-01-22         0   
1           NaN      Albania  41.153300   20.168300 2020-01-22         0   
2           NaN      Algeria  28.033900    1.659600 2020-01-22         0   
3           NaN      Andorra  42.506300    1.521800 2020-01-22         0   
4           NaN       Angola -11.202700   17.873900 2020-01-22         0   
...         ...          ...        ...         ...        ...       ...   
320035      NaN    Venezuela   6.423800  -66.589700 2023-03-09    552162   
320036      NaN      Vietnam  14.058324  108.277199 2023-03-09  11526994   
320037      NaN        Yemen  15.552727   48.516388 2023-03-09     11945   
320038      NaN       Zambia -13.133897   27.849332 2023-03-09    343135   
320039      NaN     Zimbabwe -19.015438   29.154857 2023-03-09    264276   

         dead  
0           0  
1           0  
2           0  
3           0  
4      

159

##### O total de paises validos são 159

# Questão 2

In [300]:
dados_covid_brasil = dados_covid_filtrado[dados_covid_filtrado['country'] == "Brazil"]

# print(dados_covid_brasil)
# No caso do Brazil, nãoi a diferença por provincia e todo dado é de um novo dia. logo só preciso pegar o ultimo

maximo_brasil = (max(dados_covid_brasil['date']))
print(dados_covid_brasil[dados_covid_brasil['date'] == maximo_brasil ])

       province country     Lat     Long       date  infected    dead
319790      NaN  Brazil -14.235 -51.9253 2023-03-09  37076053  699276


In [301]:
dados_covid_italia = dados_covid_filtrado[dados_covid_filtrado['country'] == "Italy"]

# print(dados_covid_italia)
# Mesmo caso para italia

maximo_italia = (max(dados_covid_italia['date']))
print(dados_covid_italia[dados_covid_italia['date'] == maximo_italia ])

       province country       Lat      Long       date  infected    dead
319911      NaN   Italy  41.87194  12.56738 2023-03-09  25603510  188322


In [302]:
dados_covid_us = dados_covid_filtrado[dados_covid_filtrado['country'] == "US"]

# print(dados_covid_italia)
# Mesmo caso para italia

maximo_us = (max(dados_covid_us['date']))
print(dados_covid_us[dados_covid_us['date'] == maximo_us ])

       province country   Lat   Long       date   infected     dead
320013      NaN      US  40.0 -100.0 2023-03-09  103802702  1123836


#### O total de casos de infecção são:

Brasil = 37076053 
Italia = 25603510
Us = 103802702

# Questão 3
 

In [303]:
# # limpas o df só com as data mais recente

maximo_df = (max(dados_covid_filtrado['date']))
print(maximo_df)
valor_data = ["2023-03-09 00:00:00"] # pura gambiarra
dados_covid_filtrado_2 = dados_covid_filtrado[dados_covid_filtrado['date'].isin(valor_data)]

# print(dados_covid_filtrado_2)
# dados_covid_filtrado_2['country'].nunique()


2023-03-09 00:00:00


C:\Users\zrkvi\AppData\Local\Temp\ipykernel_18308\2033665563.py:6: FutureWarning: The behavior of 'isin' with dtype=datetime64[ns] and castable values (e.g. strings) is deprecated. In a future version, these will not be considered matching by isin. Explicitly cast to the appropriate dtype before calling isin instead.
  dados_covid_filtrado_2 = dados_covid_filtrado[dados_covid_filtrado['date'].isin(valor_data)]


# Questão 4

In [304]:
dados_covid_filtrado_2['mortality_rate'] = dados_covid_filtrado_2.dead / dados_covid_filtrado_2.infected
print(dados_covid_filtrado_2)



       province      country        Lat        Long       date  infected  \
319760      NaN  Afghanistan  33.939110   67.709953 2023-03-09    209451   
319761      NaN      Albania  41.153300   20.168300 2023-03-09    334457   
319762      NaN      Algeria  28.033900    1.659600 2023-03-09    271496   
319763      NaN      Andorra  42.506300    1.521800 2023-03-09     47890   
319764      NaN       Angola -11.202700   17.873900 2023-03-09    105288   
...         ...          ...        ...         ...        ...       ...   
320035      NaN    Venezuela   6.423800  -66.589700 2023-03-09    552162   
320036      NaN      Vietnam  14.058324  108.277199 2023-03-09  11526994   
320037      NaN        Yemen  15.552727   48.516388 2023-03-09     11945   
320038      NaN       Zambia -13.133897   27.849332 2023-03-09    343135   
320039      NaN     Zimbabwe -19.015438   29.154857 2023-03-09    264276   

         dead  mortality_rate  
319760   7896        0.037699  
319761   3598        0.

C:\Users\zrkvi\AppData\Local\Temp\ipykernel_18308\3070830958.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dados_covid_filtrado_2['mortality_rate'] = dados_covid_filtrado_2.dead / dados_covid_filtrado_2.infected


a taxa de mortalidade da forma que foi calculada é uma péssima métrica, pois torna os valores muito pequenos. isso não causiria pânico algum durante a covid

# Questão 5